ospf6d/ospf6_auth_trailer.c

```diff
@@ -0,0 +1,817 @@
+	0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c,
+	0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c,
+	0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c,
+	0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c,
+	0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c,
+	0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c,
+	0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c,
+	0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c,
+	0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c, 0x5c,
+};
+
+void ospf6_auth_hdr_dump_send(struct ospf6_header *ospfh, uint16_t length)
+{
+	struct ospf6_auth_hdr *ospf6_at_hdr;
+	int at_len, oh_len, at_hdr_len, hash_len;
+	unsigned char temp[KEYCHAIN_MAX_HASH_SIZE+1];
+
+	oh_len = htons(ospfh->length);
+	at_len = length - oh_len;
+	if (at_len > 0) {
+		ospf6_at_hdr = (struct ospf6_auth_hdr *)
+					((uint8_t *)ospfh + oh_len);
+		at_hdr_len = htons(ospf6_at_hdr->length);
+		hash_len = at_hdr_len - OSPF6_AUTH_HDR_MIN_SIZE;
+		memcpy(temp, ospf6_at_hdr->data, hash_len);
+		temp[hash_len] = '\0';
+		zlog_debug("OSPF6 Authentication Trailer");
+		zlog_debug("  Type %d", htons(ospf6_at_hdr->type));
+		zlog_debug("  Length %d", at_hdr_len);
+		zlog_debug("  Reserved %d", ospf6_at_hdr->reserved);
+		zlog_debug("  SA ID %d", htons(ospf6_at_hdr->id));
+		zlog_debug("  seqnum high 0x%08x",
+			   htonl(ospf6_at_hdr->seqnum_h));
+		zlog_debug("  seqnum high 0x%08x",
+			   htonl(ospf6_at_hdr->seqnum_l));
+		zlog_debug("  Data %s", temp);
+	}
+}
+
+void ospf6_auth_hdr_dump_recv(struct ospf6_header *ospfh, uint16_t length)
+{
+	struct ospf6_auth_hdr *ospf6_at_hdr;
+	int at_len, oh_len, at_hdr_len, hash_len;
+	unsigned char temp[KEYCHAIN_MAX_HASH_SIZE+1];
+
+	oh_len = ntohs(ospfh->length);
+	at_len = length - oh_len;
+	if (at_len > 0) {
+		ospf6_at_hdr = (struct ospf6_auth_hdr *)
+					((uint8_t *)ospfh + oh_len);
+		at_hdr_len = ntohs(ospf6_at_hdr->length);
+		hash_len = at_hdr_len - OSPF6_AUTH_HDR_MIN_SIZE;
+		memcpy(temp, ospf6_at_hdr->data, hash_len);
+		temp[hash_len] = '\0';
+		zlog_debug("OSPF6 Authentication Trailer");
+		zlog_debug("  Type %d", ntohs(ospf6_at_hdr->type));
+		zlog_debug("  Length %d", at_hdr_len);
+		zlog_debug("  Reserved %d", ospf6_at_hdr->reserved);
+		zlog_debug("  SA ID %d", ntohs(ospf6_at_hdr->id));
+		zlog_debug("  seqnum high 0x%08x",
+			   ntohl(ospf6_at_hdr->seqnum_h));
+		zlog_debug("  seqnum high 0x%08x",
+			   ntohl(ospf6_at_hdr->seqnum_l));
+		zlog_debug("  Data %s", temp);
+	}
+}
+
+unsigned char *ospf6_hash_message_xor(unsigned char *mes1,
+				      unsigned char *mes2,
+				      uint32_t len)
+{
+	unsigned char *result;
+	uint32_t i;
+
+	result = XCALLOC(MTYPE_OSPF6_AUTH_HASH_XOR, len + 1);
+	if (!result)
+		return NULL;
+
+	for (i = 0; i < len; i++)
+		result[i] = mes1[i] ^ mes2[i];
+
+	result[len] = '\0';
```

Comment:
Why is there a `\0` being added at the end here?  This isn't a string.

Reply:
handled.